### ISSUE TITLE
RMC incorrectly reports an assertion failure on a two-dimensional vector

### ISSUE THREAD

**Issue by zhassan-aws** [model-checking/rmc#435]
I tried this code:

```rust
fn main() {
    let a: Vec<Vec<i32>> = vec![vec![0; 2]; 1];
    assert!(a.len() == 1);
}
```

using the following command line invocation:

```
rmc test.rs
```

with RMC version: `main-154-2021-08-06`

I expected the assertion to pass, but it didn't:
```
[main.assertion.1] line 3 assertion failed: a.len() == 1: FAILURE
```
If I compile the test and run it, the assertion is not hit:
```
$ rustc test.rs 
$ ./test 
$ 
```

Also, if I modify the first line in the function to
```rust
    let a: Vec<Vec<i32>> = vec![vec![0; 2]];
```
i.e. leave out the length for the outer vector, the assertion passes.

**Comment by chinmaydd** [comment]
The callstack on using the `vec!` initialization macro looks like:

[+] [`extend_with`](https://stdrs.dev/nightly/x86_64-pc-windows-gnu/src/alloc/vec/mod.rs.html#2237)
[+] [`from_elem`](https://doc.rust-lang.org/nightly/src/alloc/vec/spec_from_elem.rs.html#12)
[+] [`from_elem`](https://doc.rust-lang.org/src/alloc/macros.rs.html#47)

`extend_with` uses the `SetLenOnDrop` optimization as described in #191. Since rmc does not currently handle drop behavior, the length of the Vec does not hold the correct value after initialization. This behavior should be fixed by a combination #402 and #336.

**Comment by avanhatt** [comment]
Added as a unit test in #402
